domain/infosync/info.go

```diff
@@ -359,9 +405,31 @@ func (is *InfoSyncer) newSessionAndStoreServerInfo(ctx context.Context, retryCnt
 		return errors.Trace(err)
 	})
 	err = is.storeServerInfo(ctx)
+	if err != nil {
+		return err
+	}
+
+	logPrefix = fmt.Sprintf("[topology-syncer] %s", is.serverInfoPath)
+	session, err = owner.NewSession(ctx, logPrefix, is.etcdCli, retryCnt, TopologySessionTTL)
+	if err != nil {
+		return err
+	}
+	is.topologySession = session
+	err = is.storeTopologyInfo(ctx)
 	return err
 }
 
+// RefreshTopology refreshes etcd topology with ttl stored in "/topology/tidb/ip:port/ttl".
+func (is *InfoSyncer) RefreshTopology(ctx context.Context) error {
```

Comment:
Why `/topology/tidb/ip:port/tidb` is not refreshed? What if user deletes it?

Reply:
You're right, I think we have to ways to solve this:
* Refresh `tidb` in `RefreshTopology`.
* Watch and using `IsModify()` to check `tidb` in `topologySyncerKeeper`  
I think the second is better, but maybe `Restart` will be conflict with it.